src/kademlia/rpc_manager.cpp

```diff
@@ -300,18 +300,17 @@ bool rpc_manager::incoming(msg const& m, node_id* id
 		, total_milliseconds(now - o->sent()), print_endpoint(m.addr).c_str());
 #endif
 
+	if (m.message.dict_find_string_value("y") == "e") 
+	{
+		// it's an error.
+		o->reply(m);
+		return false;
+	}
+
 	bdecode_node ret_ent = m.message.dict_find_dict("r");
 	if (!ret_ent)
 	{
-		// it may be an error
-		ret_ent = m.message.dict_find("e");
 		o->timeout();
-		if (!ret_ent)
-		{
-			entry e;
```

Comment:
I can understand removing the dict_find("e") and ret_ent. but shouldn't the incoming_error() call still be there? just unconditionally?